### PR TITLE
BD1: LigatureLayoutBug

### PR DESCRIPTION
**Pull request by ncsouvenir**
This sample project was made in response to ticket: [UIOS-4682](https://urbnit.atlassian.net/browse/UIOS-4682).

When ligatures are added to a horizontal collection view with fixed height, it breaks the minimum inter--item spacing (horizontal spacing) and cells are placed up against one another.

However, when a horizontal collection view implements a datasource with ligatures, it will ignore the minimum inter-item spacing and default to the minimum line spacing (vertical spacing) which acts as the minimum inter--item spacing(WEIRD!)

This sample project shows examples of three horizontal collection views populated with different datasources and different minimumLineSpacing to show how the datasources with ligatures ignore the  minimum inter-item spacing.

### CODE REVIEW

File: LigatureLayoutBug/LigatureLayoutBug/ExampleCollectionView.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        self.showsHorizontalScrollIndicator = false
+
+        self.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            self.heightAnchor.constraint(equalToConstant: flowLayout.itemSize.height),
+            self.topAnchor.constraint(equalTo: self.topAnchor),
+            self.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            self.trailingAnchor.constraint(equalTo: self.trailingAnchor)
+            ])
```
Comment (JohnBueno) [inline LigatureLayoutBug/LigatureLayoutBug/ExampleCollectionView.swift]: This code is a bit confusing.  I see that you are trying to set the height of the collectionView but you are also just resetting all of the other anchors to themselves which should in theory have no effect. 

I think you could achieve your purpose with a:

`heightAnchor.constraint(equalToConstant: flowLayout.itemSize.height).isActive = true`

File: LigatureLayoutBug/LigatureLayoutBug/ExampleView.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+//
+//        cell.configureCell(with: dataSource[indexPath.row])
+//
+//        let cellSize = cell.systemLayoutSizeFitting(CGSize(width: UILayoutFittingCompressedSize.width, height: 44.0), withHorizontalFittingPriority: .fittingSizeLevel, verticalFittingPriority: .defaultHigh)
+//
+//        return cellSize
+//    }
+//}
+
+
```
Comment (JohnBueno) [inline LigatureLayoutBug/LigatureLayoutBug/ExampleView.swift]: Can this whole file just be removed?

**Review by JohnBueno — CHANGES_REQUESTED** [review]

File: LigatureLayoutBug/LigatureLayoutBug/ExampleView.swift (hunk 2 above)
Comment (ncsouvenir) [reply LigatureLayoutBug/LigatureLayoutBug/ExampleView.swift]: Yes, I just deleted it.

File: LigatureLayoutBug/LigatureLayoutBug/ExampleCollectionView.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        super.init(frame: .zero, collectionViewLayout: flowLayout) //must have designated initializer
+        
+        self.dataSource = self
+        self.delegate = self
+        self.register(LabelCollectionViewCell.self, forCellWithReuseIdentifier: "ButtonCell")
+        self.backgroundColor = .blue
+        self.showsHorizontalScrollIndicator = false
+
+        self.translatesAutoresizingMaskIntoConstraints = false
+        self.heightAnchor.constraint(equalToConstant: flowLayout.itemSize.height).isActive = true
```
Comment (ryang1428) [inline LigatureLayoutBug/LigatureLayoutBug/ExampleCollectionView.swift]: minor but you don't need all the `self.`s here (after super)

**Review by ryang1428 — CHANGES_REQUESTED** [review]

**Review by ryang1428 — APPROVED** [review]

**Review by JohnBueno — APPROVED** [review]

**Pull request merged**